src/mapml/handlers/ContextMenu.js

```diff
@@ -1027,82 +1121,91 @@ export var ContextMenu = L.Handler.extend({
-          this._items[5].el.el.innerHTML &&
-        !this._coordMenu.hasAttribute('hidden')
+          this._items[this._menuItems.CTXCOPY].el.el.innerHTML &&
+        !this._copySubMenu.hasAttribute('hidden')
       ) {
-        this._coordMenu.children[0].focus();
+        this._copySubMenu.children[0].focus();
       }
-    } else if (key === 37) {
-      //left arrow
+    } else if (e.code === 'ArrowLeft') {
       if (
-        !this._coordMenu.hasAttribute('hidden') &&
+        !this._copySubMenu.hasAttribute('hidden') &&
         document.activeElement.shadowRoot !== null
       ) {
         if (
           document.activeElement.shadowRoot.activeElement.innerHTML ===
-            this._coordMenu.children[0].innerHTML ||
+            this._copySubMenu.children[this._menuItems.CPYMENUMAP].innerHTML ||
           document.activeElement.shadowRoot.activeElement.innerHTML ===
-            this._coordMenu.children[1].innerHTML ||
+            this._copySubMenu.children[this._menuItems.CPYMENUEXTENT]
+              .innerHTML ||
           document.activeElement.shadowRoot.activeElement.innerHTML ===
-            this._coordMenu.children[2].innerHTML
+            this._copySubMenu.children[this._menuItems.CPYMENULOC].innerHTML
         ) {
-          this._coordMenu.setAttribute('hidden', '');
-          this._setActiveItem(5);
+          this._copySubMenu.setAttribute('hidden', '');
+          this._setActiveItem(this._menuItems.CTXCOPY);
         }
       }
-    } else if (key === 27) {
-      //esc key
+    } else if (e.code === 'Escape') {
+      if (this._layerMenuTabs) {
+        L.DomEvent.stop(e);
+        this._focusOnLayerControl();
+        return;
+      }
       if (document.activeElement.shadowRoot === null) {
         this._hide();
       } else {
-        if (!this._coordMenu.hasAttribute('hidden')) {
+        if (!this._copySubMenu.hasAttribute('hidden')) {
           if (
             document.activeElement.shadowRoot.activeElement.innerHTML ===
-              this._coordMenu.children[0].innerHTML ||
+              this._copySubMenu.children[this._menuItems.CPYMENUMAP]
+                .innerHTML ||
             document.activeElement.shadowRoot.activeElement.innerHTML ===
-              this._coordMenu.children[1].innerHTML ||
+              this._copySubMenu.children[this._menuItems.CPYMENUEXTENT]
+                .innerHTML ||
             document.activeElement.shadowRoot.activeElement.innerHTML ===
-              this._coordMenu.children[2].innerHTML
+              this._copySubMenu.children[this._menuItems.CPYMENULOC].innerHTML
           ) {
-            this._coordMenu.setAttribute('hidden', '');
-            this._setActiveItem(5);
+            this._copySubMenu.setAttribute('hidden', '');
+            this._setActiveItem(this._menuItems.CTXCOPY);
           }
         } else {
           this._hide();
         }
       }
+      // I don't understand the logic here, but tests break without it...
```

Comment:
I think this code block is hiding the contextmenu whenever the user presses enter on any of the layer contextmenu items (except for the "Copy (C)" item)